src/plots.py

```diff
@@ -151,27 +152,43 @@ def plot_histo(
         if layered
         else {}
     )
-
     st.vega_lite_chart(
         data=df,
         spec={
             **CONFIG_MAIN,
             "mark": {"type": "bar", **CONFIG_MARK, "blend": "normal"},
             "params": config_params,
+            "transform": [
+                {"bin": True, "field": col_x, "as": "bin_col_x", "maxbins": bin},
+                {
+                    "aggregate": [{"op": "count", "as": "xCount"}],
+                    "groupby": ["bin_col_x", "bin_col_x_end"],
+                },
+                {"joinaggregate": [{"op": "sum", "field": "xCount", "as": "TotalCount"}]},
+                {"calculate": "datum.xCount/datum.TotalCount", "as": "PercentOfTotal"},
+            ]
+            if normalize
+            else [],
             "encoding": {
                 "x": {
-                    "field": col_x,
+                    "field": "bin_col_x" if normalize else col_x,
                     "type": "ordinal" if ordinal else "quantitative",
                     "axis": {"labelAngle": -45 if ordinal else 0},
-                    "bin": config_bin,
+                    "bin": {"maxbins": bin, "binned": True} if normalize else config_bin,
```

Comment:
Labels on the x-axis should ideally be responsive to the bin slider as well (like when normalize is unchecked)

Reply:
I think so this would be resolved by the change you mentioned above.